core/src/test/java/org/elasticsearch/test/InternalTestCluster.java

```diff
@@ -1238,7 +1238,7 @@ public void restartRandomDataNode(RestartCallback callback) throws Exception {
     /**
      * Restarts a random node in the cluster and calls the callback during restart.
      */
-    private void restartRandomNode(Predicate<NodeAndClient> predicate, RestartCallback callback) throws Exception {
+    public void restartRandomNode(Predicate<NodeAndClient> predicate, RestartCallback callback) throws Exception {
```

Comment:
I needed some way to restart dedicated nodes with and without data folder. What is the preferred way to do this is? 
Also, why would it be bad to expose this?